core/fmt/fmt.odin

```diff
@@ -44,6 +44,34 @@ Info :: struct {
 // Custom formatter signature. It returns true if the formatting was successful and false when it could not be done
 User_Formatter :: #type proc(fi: ^Info, arg: any, verb: rune) -> bool
 
+// Example User Formatter:
+// SomeType :: struct {
+// 	value: int,
+// }
+// // Custom Formatter for SomeType
+// User_Formatter :: proc(fi: ^fmt.Info, arg: any, verb: rune) -> bool {
+// 	m := cast(^SomeType)arg.data
+// 	switch verb {
+// 	case 'v', 'd':
+// 		fmt.fmt_int(fi, u64(m.value), true, 32, verb)
+// 	case:
+// 		return false
+// 	}
+// 	return true
+// }
+// //
+// main :: proc() {
+// 	// Ensure the fmt._user_formatters map is initialized
+// 	if fmt._user_formatters == nil {
+// 		fmt._user_formatters = new(map[typeid]fmt.User_Formatter)
+// 	}
+// 	// Register the custom formatter directly to the _user_formatters map
```

Comment:
I agree that you can do that directly, but perhaps the example should promote using `set_user_formatters` & `register_user_formatter` instead.

Users who want to set it up directly can easily figure out how to do so by reading those procedures. What if we were to make that global private in the future?

Reply:
fixed, i also added an if guard around set_user_formatters, do you agree with that?